core/unit_test/TestViewCtorProp.hpp

```diff
@@ -0,0 +1,145 @@
+  static_assert(Kokkos::Impl::is_view_label<std::string>::value);
+
+  constexpr unsigned N = 3;
+  static_assert(Kokkos::Impl::is_view_label<const char[N]>::value);
+  static_assert(Kokkos::Impl::is_view_label<char[N]>::value);
+
+  // A char* is not a label. Thus, a label is distinguished from a pointer type.
+  static_assert(!Kokkos::Impl::is_view_label<char*>::value);
+}
+
+// Check traits of Kokkos::Impl::ViewCtorProp<>.
+TEST(TEST_CATEGORY, vcp_empty_traits) {
+  using vcp_empty_t = Kokkos::Impl::ViewCtorProp<>;
+
+  // Check that the empty view constructor properties class is default
+  // constructible. This is needed for calls of Kokkos::view_alloc().
+  static_assert(std::is_default_constructible_v<vcp_empty_t>);
+
+  static_assert(std::is_same_v<decltype(Kokkos::view_alloc()), vcp_empty_t>);
+}
+
+// Check traits of base class Kokkos::Impl::ViewCtorProp<void, std::string>.
+TEST(TEST_CATEGORY, vcp_label_base_traits) {
+  using vcp_label_base_t = Kokkos::Impl::ViewCtorProp<void, std::string>;
+
+  static_assert(std::is_same_v<typename vcp_label_base_t::type, std::string>);
+
+  // Check that the base class is default constructible. The default constructor
+  // may be called by the copy constructor of deriveded classes, such as when
+  // copy constructing a view constructor properties object from another view
+  // constructor properties object that holds fewer properties.
+  static_assert(std::is_default_constructible_v<vcp_label_base_t>);
+
+  static_assert(std::is_constructible_v<vcp_label_base_t, std::string>);
+  static_assert(std::is_constructible_v<vcp_label_base_t, const std::string&>);
+  static_assert(std::is_constructible_v<vcp_label_base_t, std::string&&>);
+
+  constexpr unsigned N = 3;
+  static_assert(std::is_constructible_v<vcp_label_base_t, const char[N]>);
+  static_assert(std::is_constructible_v<vcp_label_base_t, char[N]>);
+
+  static_assert(std::is_constructible_v<vcp_label_base_t, char*>);
+}
+
+// Check traits of derived class Kokkos::Impl::ViewCtorProp<std::string>.
+TEST(TEST_CATEGORY, vcp_label_traits) {
+  using vcp_label_base_t = Kokkos::Impl::ViewCtorProp<void, std::string>;
+  using vcp_label_t      = Kokkos::Impl::ViewCtorProp<std::string>;
+
+  static_assert(std::is_base_of_v<vcp_label_base_t, vcp_label_t>);
+
+  static_assert(vcp_label_t::has_label);
+
+  // Check that the derived class is not default constructible. It is a design
+  // choice to not allow the default constructor to be called.
+  static_assert(!std::is_default_constructible_v<vcp_label_t>);
+
+  static_assert(std::is_constructible_v<vcp_label_t, std::string>);
+  static_assert(std::is_constructible_v<vcp_label_t, const std::string&>);
+  static_assert(std::is_constructible_v<vcp_label_t, std::string&&>);
+
+  constexpr unsigned N = 3;
+  static_assert(std::is_constructible_v<vcp_label_t, const char[N]>);
+  static_assert(std::is_constructible_v<vcp_label_t, char[N]>);
+
+  static_assert(std::is_constructible_v<vcp_label_t, char*>);
+}
```

Comment:
Not finding this super useful.

Reply:
Indeed! Mostly all deleted. We kept the check of the label being stored as `std::string`, the check of `has_label`, and the check of the inheritance.